test/powershell/Modules/Microsoft.PowerShell.Diagnostics/Export-Counter.Tests.ps1

```diff
@@ -0,0 +1,216 @@
+            $importedCounterValues = Import-Counter $filePath
+
+            CompareCounterSets $counterValues $importedCounterValues
+        }
+
+        # Run a test case
+        function RunTest($testCase)
+        {
+            It "$($testCase.Name)" {
+                $getCounterParams = ""
+                if ($testCase.ContainsKey("GetCounterParams"))
+                {
+                    $getCounterParams = $testCase.GetCounterParams
+                }
+                $counterValues = &([ScriptBlock]::Create("Get-Counter -Counter `$counterNames $getCounterParams"))
+
+                # build up a command
+                $filePath = ""
+                $pathParam = ""
+                $formatParam = ""
+                if ($testCase.ContainsKey("Path"))
+                {
+                    $filePath = $testCase.Path
+                }
+                else
+                {
+                    if ($testCase.ContainsKey("FileFormat"))
+                    {
+                        $formatParam = "-FileFormat $($testCase.FileFormat)"
+                        $filePath = "$outputDirectory/$rootFilename.$($testCase.FileFormat)"
+                    }
+                    else
+                    {
+                        $filePath = "$outputDirectory/$rootFilename.blg"
+                    }
+                }
+                if ($testCase.NoDashPath)
+                {
+                    $pathParam = $filePath
+                }
+                else
+                {
+                    $pathParam = "-Path $filePath"
+                }
+                $cmd = "$cmdletName $pathParam $formatParam -InputObject `$counterValues $($testCase.Parameters) -ErrorAction Stop"
+
+                if ($testCase.CreateFirst)
+                {
+                    if (-not (Test-Path $filePath))
+                    {
+                        New-Item $filePath -ItemType file
+                    }
+                }
+
+                try
+                {
+                    if ($testCase.ContainsKey("Script"))
+                    {
+                        # Here we want to run the command then do our own post-run checks
+                        $sb = [ScriptBlock]::Create($cmd)
+                        &$sb
+                        &$testCase.Script
+                    }
+                    else
+                    {
+                        # Here we expect and want the command to fail
+                        try
+                        {
+                            $sb = [ScriptBlock]::Create($cmd)
+                            &$sb
+                            throw "Did not throw expected exception"
+                        }
+                        catch
+                        {
+                            $_.FullyQualifiedErrorId | Should Be $testCase.ExpectedErrorId
+                        }
+                    }
+                }
+                finally
+                {
+                    if ($filePath)
```

Comment:
This test was not meant to determine if a file exists, just whether the $filePath variable contains a path, to prevent invoking Remove-Item with a null or empty path. I can remove the test if you like.